pkg/server/api/svid/v1/service_test.go

```diff
@@ -0,0 +1,246 @@
+package svid_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api/rpccontext"
+	"github.com/spiffe/spire/pkg/server/api/svid/v1"
+	"github.com/spiffe/spire/pkg/server/ca"
+	"github.com/spiffe/spire/test/clock"
+	"github.com/spiffe/spire/test/fakes/fakeserverca"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/spiffe/spire/test/testkey"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+)
+
+func TestServiceMintX509SVID(t *testing.T) {
+	clock := clock.NewMock(t)
+	// Add logger to context
+	log, logHook := test.NewNullLogger()
+	ctx := rpccontext.WithLogger(context.Background(), log)
+
+	spiffeID := spiffeid.Must("example.org", "workload1")
+	dns := []string{"dns1", "dns2"}
+
+	// Create Service
+	trustDomain := spiffeid.RequireTrustDomainFromString("example.org")
+	fakeServerCA := fakeserverca.New(t, trustDomain.String(), &fakeserverca.Options{Clock: clock})
+	service := svid.New(svid.Config{
+		ServerCA:    fakeServerCA,
+		TrustDomain: trustDomain,
+	})
+
+	key := testkey.NewEC256(t)
+	x509CA := fakeServerCA.X509CA()
+	now := fakeServerCA.Clock().Now().UTC()
+
+	testCases := []struct {
+		name      string
+		code      codes.Code
+		createCsr func(tb testing.TB) *x509.CertificateRequest
+		dns       []string
+		err       string
+		expiredAt time.Time
+		msg       string
+		spiffeID  spiffeid.ID
+		subject   string
+		ttl       time.Duration
+		x509CA    *ca.X509CA
+	}{
+		{
+			name: "success",
+			createCsr: func(tb testing.TB) *x509.CertificateRequest {
+				template := &x509.CertificateRequest{
+					SignatureAlgorithm: x509.ECDSAWithSHA256,
+					URIs:               []*url.URL{spiffeID.URL()},
+					DNSNames:           dns,
+					Subject: pkix.Name{
+						Country:      []string{"US", "EN"},
+						Organization: []string{"ORG"},
+					},
+				}
+				return createCsr(t, template, key)
+			},
+			dns:       []string{"dns1", "dns2"},
+			expiredAt: now.Add(time.Minute),
+			spiffeID:  spiffeID,
+			subject:   "CN=dns1,O=ORG,C=US+C=EN",
+			ttl:       time.Minute,
+			x509CA:    x509CA,
+		}, {
+			name: "default values",
```

Comment:
are there other "default values" being tested here other than TTL? If not, maybe this test should be changed to test that we can override the TTL with the request and the rest of the tests can use the default. The benefit is that this would be the only test case that has to set the `ttl` field on the test case instead of the other way around.

Reply:
iit is testing subject as well, and no dns is returned